tests/integration/ie-tests.sh

```diff
@@ -46,21 +54,30 @@ test_example_job() {
   pi=$(kubectl --kubeconfig=${KUBE_CONFIG} logs $(kubectl --kubeconfig=${KUBE_CONFIG} get pods | tail -n 1 | cut -d' ' -f1)  | grep 'Pi is roughly' | rev | cut -d' ' -f1 | rev | cut -c 1-3)
   echo -e "Spark Pi Job Output: \n ${pi}"
 
-  spark-client.service-account-registry delete --username=ie-test
-
-  account_deleted=$(spark-client.service-account-registry get-conf --username=ie-test 2>&1 | grep -c NoAccountFound)
-
   if [ "${pi}" != "3.1" ]; then
       exit 1
   fi
 
+}
+
+test_example_job() {
+  spark-client.service-account-registry delete --username=ie-test
+  spark-client.service-account-registry create --username=ie-test
+
+  run_example_job default ie-test
+
+  spark-client.service-account-registry delete --username=ie-test
+
+  account_deleted=$(spark-client.service-account-registry get-conf --username=ie-test 2>&1 | grep -c NotFound)
+
   if [ "${account_deleted}" == "0" ]; then
       exit 1
   fi
 
 }
 
 test_spark_shell() {
+  spark-client.service-account-registry delete --username=ie-test
```

Comment:
In case of unexpected test failures, proper cleanup of username does not always happen.

For repeated runs, I put this in.

Reply:
Uhm, I see. But maybe I would make this as a utility function `cleanup`, that is called at the end of all tests, regardless of the outcome of the tests and that can also be called manually. 

I believe that deleting things first is departing a bit from having a nice clean test setup. Also, I believe we should rather perform a check at the beginning just to make sure we are not overriding stuff that already exists, in that case we should just stop the tests

It is not a super important point, but I believe it would make things better TBH